sdk/go/common/resource/plugin/host.go

```diff
@@ -118,6 +113,11 @@ func IsLocalPluginPath(source string) bool {
 		return true
 	}
 
+	if pluginSpec.IsGitPlugin() {
+		// If it's a git plugin, it's not a local path
+		return false
+	}
+
 	// If there is a download URL or the name matches the plugin name regexp after parsing, it's not a local path
 	return pluginSpec.PluginDownloadURL == "" && !workspace.PluginNameRegexp.MatchString(pluginSpec.Name)
```

Comment:
Separately, one wonders if there's grounds to bundle this up into a method on the spec too.